cbor/shared/src/main/scala/fs2/data/cbor/high/internal/ValueParser.scala

```diff
@@ -209,16 +208,31 @@ object ValueParser {
   private def parseTags[F[_]](chunk: Chunk[CborItem],
                               idx: Int,
                               rest: Stream[F, CborItem],
-                              tags: CborValue => CborValue,
+                              tags: CborValue => Pull[F, Nothing, CborValue],
                               chunkAcc: List[CborValue])(implicit
-      F: RaiseThrowable[F]): Pull[F, CborValue, Result[F, CborValue => CborValue]] =
+      F: RaiseThrowable[F]): Pull[F, CborValue, Result[F, CborValue => Pull[F, Nothing, CborValue]]] =
     if (idx >= chunk.size) {
       Pull.output(Chunk.seq(chunkAcc.reverse)) >> rest.pull.uncons.flatMap {
         case Some((hd, tl)) => parseTags(hd, 0, tl, tags, Nil)
         case None           => Pull.raiseError(new CborParsingException("unexpected end of input"))
       }
     } else {
       chunk(idx) match {
+        case CborItem.Tag(Tags.PositiveBigNum) =>
+          def f: CborValue => Pull[F, Nothing, CborValue] = {
+            case CborValue.ByteString(bytes) => tags(CborValue.Integer(BigInt(bytes.toArrayUnsafe)))
+            case CborValue.Tagged(tag2, v)   => f(v).map(CborValue.Tagged(tag2, _))
+            case _ => Pull.raiseError(new CborTagDecodingException("A bignum must have a byte string as value"))
```

Comment:
What about Indefinite [Length Byte String](https://www.rfc-editor.org/rfc/rfc8949.html#name-indefinite-length-byte-stri)? They are byte strings after all, and this would be valid as an encoding I guess.

Reply:
This is true, I misread, indeed.